### PR TITLE
Fixed single arg flaw with string types.

### PR DESCRIPTION
**Pull request by anywhichway**
Your current code is subject to this flaw: https://codeburst.io/a-key-to-keys-when-javascript-keys-dont-match-ab44c81adc87

The pull request fixes the flaw with marginal impact on performance.

### CODE REVIEW

**Comment by anywhichway** [comment]
This probably failed your build because I added a unit test that will fail with your legacy code.

**Comment by anywhichway** [comment]
This pull request includes the previous pull request to fix the string key flaw. It also more than doubles the speed of fast-memoize in the single argument case. My guess is you will get a jump in multi-arg as well, but I don't thoroughly understand your benchmark suites, so I will leave it up to you to check.

**Comment by megawac** [comment]
This change broke my use of `fast-memoize` with a custom cache with a `WeakMap` as I implemented my checks to see whether the value exists using `has` and now `has` does not get called. It also makes the example superfluous as there is no need to implement a has method there.

File: src/index.js [hunk 1]
```diff
@@ -32,26 +32,26 @@ function isPrimitive (value) {
 function monadic (fn, cache, serializer, arg) {
   var cacheKey = isPrimitive(arg) ? arg : serializer(arg)
 
-  if (!cache.has(cacheKey)) {
-    var computedValue = fn.call(this, arg)
+  var computedValue = cache.get(cacheKey);
+  if (typeof computedValue==="undefined") {
```
Comment (megawac) [inline src/index.js]: This seems to be sacrificing API correctness for speed.

Comment (anywhichway) [reply src/index.js]: @megawac I would say mots of my path does sacrifice API correctness for speed, but the stated goal of the library is to provide the "fastest possible memoization library in JavaScript that supports N arguments." I was just working in concert with its stated goal. Also, the failure to treat string keys correctly was in my opinion a fundamental semantic/functional flaw. Take a look at: https://codeburst.io/a-key-to-keys-when-javascript-keys-dont-match-ab44c81adc87.

**Pull request merged**